src/Blocks/BlockMobSpawner.h

```diff
@@ -71,8 +71,8 @@ class cBlockMobSpawnerHandler final :
 		}
 
 		const auto Player = static_cast<const cPlayer *>(a_Digger);
-		auto Handler = Player->GetEquippedItem().GetHandler();
-		if (!Player->IsGameModeSurvival() || !Handler->CanHarvestBlock(E_BLOCK_MOB_SPAWNER))
+		auto & Handler = Player->GetEquippedItem().GetHandler();
```

Comment:
You use `Player->GetEquippedItem().GetHandler()` here, but `cItemHandler::For(Player->GetEquippedItem().m_ItemType)` below, and `cItemHandler::For(m_Player->GetEquippedItem())` in the next file down. It would look nicer if there was some consistency in how we're doing this - why's it even possible to do it these three ways in the first place!

Reply:
I'd say we can use convenience accessors on cItem when we have an instance, while For is for when we only have a bare item type.

Reply:
@tigerw That seems like a reasonable distinction to make. In that case should we remove the overload for `For` that takes a cItem? But we can certainly convert the uses in this PR to be consistent.

@12xx12 that method does seem workable, since constructing a cItem from an item id is simple, but that's still a bit more overhead than just calling `For` directly. Still an option though.

Reply:
I can't think of a situation where you would need a Item handler whiteout having a cItem. That's why I would hide the function from everything else

Reply:
It does seem like all the instances you found in this PR have cItems, and it's not exported to the API. Go ahead with the removal then!